tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -771,16 +771,22 @@ Status StorageManagerCanonical::array_get_non_empty_domain_var_from_name(
       "' does not exist"));
 }
 
-Status StorageManagerCanonical::array_get_encryption(
-    const ArrayDirectory& array_dir, EncryptionType* encryption_type) {
-  const URI& uri = array_dir.uri();
-
+Status StorageManagerCanonical::array_get_encryption(const URI& uri, EncryptionType* encryption_type) {
   if (uri.is_invalid()) {
     return logger_->status(Status_StorageManagerError(
         "Cannot get array encryption; Invalid array URI"));
   }
 
-  const URI& schema_uri = array_dir.latest_array_schema_uri();
+  // Load URIs from the array directory
+  optional<tiledb::sm::ArrayDirectory> array_dir;
+  array_dir.emplace(
+      resources_,
+      uri,
+      0,
+      UINT64_MAX,
+      tiledb::sm::ArrayDirectoryMode::SCHEMA_ONLY);
+  
+  const URI& schema_uri = array_dir->latest_array_schema_uri();
```

Comment:
```suggestion
  // Load URIs from the array directory
  ArrayDirectory array_dir(
      resources_,
      uri,
      0,
      UINT64_MAX,
      tiledb::sm::ArrayDirectoryMode::SCHEMA_ONLY);
  
  const URI& schema_uri = array_dir.latest_array_schema_uri();
```

(might not be formatted properly)